backend/inspirehep/records/tasks.py

```diff
@@ -43,3 +54,66 @@ def update_records_relations(uuids):
 
     db.session.commit()
     return uuids
+
+
+@shared_task(
+    bind=True,
+    queue="indexer_task",
+    acks_late=True,
+    retry_backoff=True,
+    retry_kwargs={"max_retries": 3},
+    autoretry_for=(OperationalError, TransportError),
+)
+def redirect_references_to_merged_record(self, uuid):
+    record = InspireRecord.get_record(uuid)
+    deleted_record_refs = record.get_new_deleted_records()
+    if deleted_record_refs:
+        record_ref = get_value(record, "self.$ref")
+        record_schema = PidStoreBase.get_schema_name_from_uri(record["$schema"])
+        possible_refs_to_record = get_refs_to_schemas()[record_schema]
+        for deleted_record_ref in deleted_record_refs:
+            update_references_pointing_to_merged_record(
+                possible_refs_to_record, deleted_record_ref, record_ref
+            )
+
+
+def update_references_pointing_to_merged_record(
+    refs_to_schema, merged_record_uri, new_record_uri
+):
+    for index, path in refs_to_schema:
+        config = get_config_for_given_path(index, path)
+        matched_records = list(match({"$ref": merged_record_uri}, config,))
```

Comment:
You can iterate on the iterator returned by `match` directly:
```suggestion
        matched_records = match({"$ref": merged_record_uri}, config,)
```